src/cache/cache.rs

```diff
@@ -203,9 +203,31 @@ pub trait Storage: Send + Sync {
 
 /// Get a suitable `Storage` implementation from the environment.
 pub fn storage_from_environment() -> Box<Storage> {
+    let endpoint_env_res = env::var("SCCACHE_ENDPOINT");
+    let mut endpoint = String::from("s3.amazonaws.com");
+    match endpoint_env_res {
+        Ok(ep) => endpoint = ep,
+        Err(e) => warn!("Failed to interpret endpoint: {:?}", e),
+    }
+    let region_env_res = env::var("SCCACHE_REGION");
```

Comment:
I disagree. It's one more configuration value, which is a pain to manage, and it's possible for it to be out-of-sync with the bucket name (I can imagine someone copy/pasting and editing these values and not changing the region properly.)